envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -216,6 +216,10 @@ message HttpConnectionManager {
     // XFCC header comma separated from other values with the value Cert="PEM".
     // Defaults to false.
     bool cert = 3;
+
+    // [#not-implemented-hide:] Whether to forward the DNS SAN entries of the client cert. Defaults
+    // to false.
+    bool dns_san = 4;
```

Comment:
Does this flag interact with the existing `san` flag? If so, please add some description of the interaction.

Reply:
It won't interact with the san flag. The san flag adds the URI type SAN, this flag will add the DNS san entries directly to XFCC header like:

...Hash=468ed33be74eee6556d90c0149c1309e9ba61d6425303443c0748a02dd8de688;SAN=https://example.com;DNS=example.com;DNS=www.example.com...

Reply:
Can you clarify the docs on the `san` field? Thanks.

Reply:
Sorry, didn't provide a detail explanation earlier. 

As per [RFC 2459](https://tools.ietf.org/html/rfc2459#section-4.2.1.7), the SAN can have multiple values and different types in the certificate.
```
SubjectAltName ::= GeneralNames

      GeneralNames ::= SEQUENCE SIZE (1..MAX) OF GeneralName

      GeneralName ::= CHOICE {
           otherName                       [0]     OtherName,
           rfc822Name                      [1]     IA5String,
           dNSName                         [2]     IA5String,
           x400Address                     [3]     ORAddress,
           directoryName                   [4]     Name,
           ediPartyName                    [5]     EDIPartyName,
           uniformResourceIdentifier       [6]     IA5String,
           iPAddress                       [7]     OCTET STRING,
           registeredID                    [8]     OBJECT IDENTIFIER}
```
As of now envoy only identifies and forwards the URI type of SAN and ignores DNS, IP, etc. even if they are present in the certificate. DNS SAN is quite useful in mTLS. [RFC2818](https://tools.ietf.org/html/rfc2818#section-3.1) for reference on how DNS SAN overrides CN.

With the `dns_san` flag, we would allow envoy configuration to append the DNS SAN entries as well in the x-forwarded-client-cert header to allow the clients to validate TLS connections based on DNS entries.

The XFCC header before:
`...Hash=468ed33be74eee6556d90c0149c1309e9ba61d6425303443c0748a02dd8de688;SAN=https://example.com;...`

and after the change, if some config enables this flag:
`...Hash=468ed33be74eee6556d90c0149c1309e9ba61d6425303443c0748a02dd8de688;SAN=https://example.com;DNS=example.com;DNS=www.example.com...`

Reply:
@shakti-das can just potentially capture some of this in the docs? I think users will be confused with the different SAN settings with the docs as is. Thank you!